### ISSUE TITLE
credential_process doesn't work when Okta MFA is required

### ISSUE THREAD

**Issue by Diaz-Manuel** [Versent/saml2aws#939]
When executing saml2aws through awscli with the config:
```
[profile dev]
credential_process = saml2aws login --skip-prompt  --quiet --credential-process --role arn:aws:iam::123412341234:role/dev --profile dev
```
It won't work since survey asks for the MFA token and `awscli` captures all stdout. Pinentry prompter (`--prompter pinentry`) should be able to work around this issue by prompting through the tty instead of stdout, but current pinentry implementation ends up calling survey in the background as the default prompter.

Pinentry's `StringRequired` ([link to function](https://github.com/Versent/saml2aws/blob/0d135dc19e45548630fb7a9a742d7d479f1c061c/pkg/prompter/pinentry.go#L71)) should be changed from:
```
func (p *PinentryPrompter) StringRequired(pr string) string {
	return p.DefaultPrompter.StringRequired(pr)
}
```
To something like:
```
func (p *PinentryPrompter) StringRequired(pr string) string {
	return p.RequestSecurityCode(pr)
}
```
Since the `DefaultPrompter` points to survey, even when `--prompter pinentry` flag is set.

**Comment by jaklan** [comment]
Struggle with the same issue. Tried to apply the workaround described here:
https://github.com/boto/botocore/issues/1348#issuecomment-579317535
but unfortunately - `saml2aws` writes not only credentials to stdout, so it fails anyway. To make it work, `saml2aws` could write everything except credentials  to `stderr` (at least when using `--credential-process` flag).

**Comment by Diaz-Manuel** [comment]
@jaklan as a workaround while they (hopefully) work on the issue, this is what I did to get it working:
1. Modify the funcion mentioned above so that when using pinentry as prompter, it asks for MFA code smoothly.
2. Set the config for your AWS profile in ~/.aws/config like this
```
[profile dev]
credential_process = sh -c 'saml2aws login --skip-prompt --prompter pinentry --credential-process --role arn:aws:iam::123412341234:role/dev --profile dev 2>$(tty)'
```
Do take into consideration that if you are using AWS from anywhere that isn't a TTY, the `tty` command would return `not a tty` through stdout. In that case you won't be able to see the errors of saml2aws, but other than that it works ok. Also, while trying to redirect output like `2>not a tty` the behaviour depends on the shell, some just fail, zsh actually creates the files `not`, `a` and `tty` in your current directory (this might be solved by changing it to `2>/dev/tty`, haven't checked.

**Comment by jaklan** [comment]
@Diaz-Manuel that's actually the same workaround I mentioned in my comment - and it doesn't work for me, also with `pinentry` prompter (MacOS, zsh). No prompt is displayed after the line `Authenticating as <username> ...` and I can only kill the command then.

**Comment by Diaz-Manuel** [comment]
Yeah, it's the same thing, just wanted to be extra clear on the arguments for saml2aws to prevent it from sending logs to stdout.
About pinentry, did you re-compile saml2aws after your changes and added it to your PATH?

**Comment by jaklan** [comment]
>About pinentry, did you re-compile saml2aws after your changes and added it to your PATH?

Ach you meant to run it after applying your code change, that clarifies a lot 😉 I tested only the unmodified package, without your fix.
